djangoplicity/archives/views.py

```diff
@@ -811,3 +813,117 @@ def has_field(model, field):
             qs = qs.filter(release_date__lte=datetime.now())
 
         return qs
+
+
+def resource_proxy_view(request, model, format, id, ext):
+    from djangoplicity.archives.options import ArchiveOptions
+    from djangoplicity.media.consts import MEDIA_CONTENT_SERVERS
+    from djangoplicity.contentserver.base import S3ContentServer
+
+    print(f"Resource proxy view: {model}, {format}, {id}, {ext}")
+
+    # Get model class
+    model_class = _get_model_class(model)
+    if not model_class:
+        return HttpResponseNotFound("Model not found")
+    
+    # Get object
+    obj = _get_object(model_class, id)
+    if not obj:
+        return HttpResponseNotFound("Object not found")
+
+    # Get options
+    options = ArchiveOptions()
+    has_access, reason = _has_access_permissions(request, obj, options)
+    
+    if not has_access:
+        return HttpResponseForbidden(reason)
+    
+    # Get content server
+    content_server = MEDIA_CONTENT_SERVERS[obj.content_server]
+    if not content_server:
+        return HttpResponseForbidden("Content server not found.")
+    if not isinstance(content_server, S3ContentServer):
```

Comment:
Validate the existence of the get_signed_url function instead of the type of the content server